app/controllers/application_controller.rb

```diff
@@ -5,6 +5,81 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter secret"
   end
+  
+  get '/' do 
+  	erb :index
+  end
+  
+  get '/signup' do 
+  	if logged_in? 
+  		redirect '/tweets'
+  	else
+			erb :'users/create_user'
+		end
+	end
+	
+	post '/signup' do 
+		user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+		if user.save && valid_user?(user)
```

Comment:
Great job using ActiveRecord model validations! However, I think your `valid_user?` method might be redundant.